kafkacat.c

```diff
@@ -628,6 +645,43 @@ static void kafkaconsumer_run (FILE *fp, char *const *topics, int topic_cnt) {
 }
 #endif
 
+/**
+ * Get offsets from conf.starts for consumer_run
+ */
+static int64_t* get_offsets(rd_kafka_metadata_topic_t *topic) {
+        int i;
+        rd_kafka_resp_err_t err;
+        rd_kafka_topic_partition_list_t *rktparlistp =
+                                           rd_kafka_topic_partition_list_new(1);
+
+        for (i = 0 ; i < (*topic).partition_cnt ; i++) {
+                int32_t partition = (*topic).partitions[i].id;
```

Comment:
`topic->partitions..`

Reply:
I write C once a year, but I was creative this time